example/src/main/java/com/xwray/groupie/example/ExpandableHeaderItem.kt

```diff
@@ -13,6 +13,8 @@ class ExpandableHeaderItem(@StringRes titleStringResId: Int,
                            @StringRes subtitleResId: Int)
     : HeaderItem(titleStringResId, subtitleResId), ExpandableItem {
 
+    var clickListener: ((ExpandableHeaderItem) -> Unit)? = null
```

Comment:
this clickListener can be an optional last parameter in the constructor (`private val`)

Reply:
I had it that way initially - but you sort of run into a chicken and the egg situation. The header wants its listener setup in the constructor, but the listener needs the group to be able to add and remove. The group can only be created with the header. I could make it a `private var` in my constructor and keep setting it the same way? Thoughts?

Reply:
ah, you're right - didn't notice that. Nah, keep it as is